keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -82,7 +88,10 @@ class ArtifactListener(
       if (latestVersion != null) {
         var status = artifactSupplier.getReleaseStatus(latestArtifact)
         log.debug("Storing latest version {} (status={}) for registered artifact {}", latestVersion, status, artifact)
-        repository.storeArtifact(artifact.name, artifact.type, latestVersion, status)
+
+        val artifactMetadata = artifactMetadata(latestArtifact)
```

Comment:
I think we should save the artifact, then fetch the metadata, then then re-save. I think this because theres a potential that those metadata calls could fail, and that shouldn’t lead to a failure to save the artifact (which would mean no one would be able to deploy that version == bad). it should just lead to less details (and we could probably show this in the UI somehow - maybe by recording on the artifact version if it’s pre or post metadata fetch?)

Reply:
Done!